src/MarginTrading.Backend.Services/Services/OvernightSwapService.cs

```diff
@@ -197,95 +188,120 @@ public void CalculateAndChargeSwaps()
-				auditLog: calculation.ToJson());
-			
-			//update calculation state if previous existed
-			var newCalcState = lastCalcExists
-				? OvernightSwapCalculation.Update(calculation, lastCalc)
-				: OvernightSwapCalculation.Create(calculation);
+            //check if swaps had already been taken
+            var lastCalcExists = _overnightSwapCache.TryGet(OvernightSwapCalculation.GetKey(account.Id, order.Instrument, order.GetOrderType(), order.Id),
+                                     out var lastCalc)
+                                 && lastCalc.Time >= CalcLastInvocationTime();
+            if (lastCalcExists)
+            {
+                await _log.WriteErrorAsync(nameof(OvernightSwapService), nameof(ProcessOrder),
+                    new Exception($"Overnight swap had already been taken, filtering: {JsonConvert.SerializeObject(lastCalc)}"), DateTime.UtcNow);
 
-			//add to cache
-			_overnightSwapCache.AddOrReplace(newCalcState);
-			
-			//write state and log
-			await _overnightSwapStateRepository.AddOrReplaceAsync(newCalcState);
-			await _overnightSwapHistoryRepository.AddAsync(calculation);
-		}
+            }
 
-		/// <summary>
-		/// Log failed orders.
-		/// </summary>
-		/// <param name="orders"></param>
-		/// <param name="clientId"></param>
-		/// <param name="accountId"></param>
-		/// <param name="instrument"></param>
-		/// <param name="exception"></param>
-		/// <returns></returns>
-		private async Task ProcessFailedOrders(IReadOnlyList<Order> orders, string clientId, string accountId, 
+            //calc swaps
+            var swapRate = order.GetOrderType() == OrderDirection.Buy ? accountAssetPair.OvernightSwapLong : accountAssetPair.OvernightSwapShort;
+            if (swapRate == 0)
+                return;
+
+            var total = _commissionService.GetOvernightSwap(order, swapRate);
+            if (total == 0)
+                return;
+
+            //create calculation obj & add to cache
+            var volume = order.Volume;
+            var calculation = OvernightSwapCalculation.Create(account.ClientId, account.Id, order.Instrument,
+                order.Id, _currentStartTimestamp, true, null, volume, total, swapRate, order.GetOrderType());
+
+            //charge comission
+            var instrumentName = _assetPairsCache.GetAssetPairByIdOrDefault(accountAssetPair.Instrument)?.Name
+                                 ?? accountAssetPair.Instrument;
+            await _accountManager.UpdateBalanceAsync(
+                account: account,
+                amount: total,
+                historyType: AccountHistoryType.Swap,
+                comment: $"{instrumentName} {(order.GetOrderType() == OrderDirection.Buy ? "long" : "short")} swaps. Volume: {volume}. Positions count: {1}. Rate: {swapRate}. Time: {_currentStartTimestamp:u}.",
```

Comment:
Comment should be changed to : Position {positionId} swaps. Time {_currentStartTimestamp:u}.